BJS2-94343 Solution of the task

KuaNySH88 opened the pull request.

KuaNySH88 committed:
BJS2-94343 Solution of the task

MarselYsup commented on src/main/java/school/faang/bjs2_94343/MasterCardService.java:
```diff
@@ -0,0 +1,47 @@
+    private static int collectPayment() {
+        try {
+            Thread.sleep(TEN_SECONDS_IN_MS);
+            return 5_000;
```
давай эти значения тоже вынесем в константы, дабы код был более гибким

KuaNySH88 replied:
поправил

MarselYsup commented on src/main/java/school/faang/bjs2_94343/MasterCardService.java:
```diff
@@ -0,0 +1,47 @@
+    private static int sendAnalytics() {
+        try {
+            Thread.sleep(ONE_SECOND_IN_MS);
+            return 17_000;
```
давай эти значения тоже вынесем в константы, дабы код был более гибким

KuaNySH88 replied:
поправил

MarselYsup commented on src/main/java/school/faang/bjs2_94343/MasterCardService.java:
```diff
@@ -0,0 +1,47 @@
+        Future<Integer> collectPaymentFuture = executorService.submit(MasterCardService::collectPayment);
+
+        CompletableFuture<Integer> sendAnalyticsFuture = CompletableFuture
+                .supplyAsync(MasterCardService::sendAnalytics);
```
Использование пула потоков по умолчанию (ForkJoinPool.commonPool) не рекомендуется в производственном коде, так как вы не контролируешь его поведение. Передавай свой собственный пул потоков в качестве аргумента.
 
 ```java
  CompletableFuture<Integer> sendAnalyticsFuture = CompletableFuture
                .supplyAsync(MasterCardService::sendAnalytics, executorService );
```

KuaNySH88 replied:
поправил

KuaNySH88 committed:
BJS2-94343 Fix PR comment:
 1. use own executorService instead of default one
 2. create new constants for return values
